ui-trackoccupancydiagram/src/components/helpers/drawElements/drawTrack.ts

```diff
@@ -0,0 +1,81 @@
+import { sum } from 'lodash';
+
+import { TRACK_HEIGHT_CONTAINER, COLORS, TICKS_PATTERN } from '../../consts';
+import { getTickPattern } from '../../utils';
+
+const { WHITE_50, GREY_20, RAIL_TICK } = COLORS;
+
+const drawRails = ({
+  xStart,
+  yStart,
+  width,
+  stroke = GREY_20,
+  ctx,
+}: {
+  xStart: number;
+  yStart: number;
+  width: number;
+  stroke?: string;
+  ctx: CanvasRenderingContext2D;
+}) => {
+  ctx.clearRect(xStart, yStart, width, 9);
+  ctx.beginPath();
+  ctx.fillStyle = WHITE_50;
+  ctx.strokeStyle = stroke;
+  ctx.lineWidth = 1;
+  ctx.rect(xStart, yStart, width, 8);
+  ctx.fill();
+  ctx.stroke();
+};
+
+const drawTick = ({
+  ctx,
+  xStart,
+  yStart,
+  ticks,
+  stroke,
+}: {
+  ctx: CanvasRenderingContext2D;
+  xStart: number;
+  yStart: number;
+  ticks: number[];
+  stroke: string;
+}) => {
+  ctx.beginPath();
+  ctx.setLineDash(ticks);
+  ctx.strokeStyle = stroke;
+  ctx.lineWidth = 1;
+  ctx.moveTo(xStart, yStart - sum(ticks) / 2);
```

Comment:
i think you can calculate sum(ticks) only one time and pass the result as a const

Reply:
done

Reply:
done